packages/@aws-cdk/aws-fsx/README.md

```diff
@@ -17,8 +17,182 @@
+Linux operating system. It also provides read-after-write consistency and supports file locking.
+
+### Installation
+
+Import to your project:
 
 ```ts
 import fsx = require('@aws-cdk/aws-fsx');
 ```
+
+### Basic Usage
+
+Setup required properties and create:
+
+```ts
+const stack = new Stack(app, 'Stack');
+const vpc = new Vpc(stack, 'VPC');
+
+const fileSystem = new FsxLustreFileSystem(stack, 'FsxLustreFileSystem', {
+  lustreConfiguration: { deploymentType: FsxLustreDeploymentType.SCRATCH_2 },
+  storageCapacity: 1200,
+  vpc,
+  vpcSubnet: vpc.privateSubnets[0]});
+```
+
+### Connecting
+
+To control who can access the file system, use the `.connections` attribute. FSx has a fixed default port, so you don't
+need to specify the port. This example allows an EC2 instance to connect to a file system:
+
+```ts
+fileSystem.connections.allowDefaultPortFrom(instance);
+```
+
+### Mounting
+
+To mount an FSx for Lustre file system programmatically, the mount name of the file system needs to be determined. The
+easiest way to do that is with the AWS CLI, which can be written into a User Data script to be run on start-up of your
+instance. The following example can be used for reference on how to bring up a file system and EC2 instance with the
+file system mounted on the instance at start-up:
+
+```ts
+const app = new App();
+const stack = new Stack(app, 'AwsCdkFsxLustre');
+const vpc = new Vpc(stack, 'VPC');
+
+// Create a security group to share across the file system and ec2 instance and open the FSx ports.
+const securityGroup = new SecurityGroup(stack, 'FsxLustreSecurityGroup', {
+  vpc
+});
+securityGroup.addIngressRule(Peer.anyIpv4(), Port.tcp(988), 'Default FSx port ingress');
+securityGroup.addEgressRule(Peer.anyIpv4(), Port.tcp(988), 'Default FSx port egress');
+
+const fs = new FsxLustreFileSystem(stack, 'FsxLustreFileSystem', {
+  lustreConfiguration: { deploymentType: FsxLustreDeploymentType.SCRATCH_2 },
+  securityGroup,
+  storageCapacity: 1200,
+  vpc,
+  vpcSubnet: vpc.privateSubnets[0]});
+
+const inst = new Instance(stack, 'Ec2Instance', {
+  instanceType: InstanceType.of(InstanceClass.T2, InstanceSize.LARGE),
+  machineImage: new AmazonLinuxImage({
+    generation: AmazonLinuxGeneration.AMAZON_LINUX_2
+  }),
+  securityGroup,
+  vpc,
+  vpcSubnets: {
+    subnetType: SubnetType.PUBLIC,
+  }
+});
+
+// Need to give the instance access to read information about FSx to determine the file system's mount name.
+inst.role.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName("AmazonFSxReadOnlyAccess"));
+
+// Add a dependency on the file system to the instance so that we can be sure the file system is instantiated before querying
+// for its mount name.
+inst.node.addDependency(fs);
+
+const mountPath = '/mnt/fsx';
+const dnsName = fs.dnsName;
+
+inst.userData.addCommands(
+  'set -eux',
+  'yum update -y',
+  'amazon-linux-extras install -y lustre2.10',
```

Comment:
The lustre version will change over time. Maybe worth thinking how you would want to version control you cdk to meet with the lustre version you need.

Reply:
This is example code in a README, not managed on behalf of the user. 

Looks like the current iteration of this is punting the version management to the user.

Is there a way the latest version could be automatically retrieved, which we can recommend or make available to users instead?

Reply:
I tried looking into it a little more but couldn't find any documentation about why 2.10 is being used rather than a more recent version. There's also no way to tie it to the latest version, unless `amazon-linux-extras` ins't used. This will at least always pick up the most recent release of 2.10.

Reply:
I got the following information about why AL2 is distributing Lustre 2.10:

> FSx for Lustre currently serves only Lustre 2.10 and we only recommend Lustre 2.10 clients for our customers. The Lustre 2.10 kernel modules are baked directly into the AL2 kernel, which is why we currently only vend Lustre 2.10 in the amazon-linux-extras repo (only the Lustre 2.10 userspace tools are included in the Lustre2.10 package in the repo).
> I think FSx for Lustre may be unique here. Using a Lustre 2.12 client, for example, may give you trouble when using FSx for Lustre backed by a 2.10 server.